eclipse-collections-code-generator/src/main/resources/impl/lazy/collectPrimitiveIterable.stg

```diff
@@ -118,12 +118,6 @@ public class Collect<name>Iterable\<T>
         return this.iterable.allSatisfy((T each) -> predicate.accept(this.function.<type>ValueOf(each)));
     }
 
-    @Override
-    public boolean noneSatisfy(final <name>Predicate predicate)
-    {
-        return this.iterable.allSatisfy((T each) -> !predicate.accept(this.function.<type>ValueOf(each)));
-    }
-
```

Comment:
All other implementations of none are !any(predicate). This one is all(!predicate). That implies any and all can be written in terms of each other too.

any --> !all(!predicate)
all --> !any(!predicate)

Is that right?

Reply:
This makes sense. The difference is that none can be defined with a single ! on the outside when using any. Quite a while ago we changed the basic implementations of all three to use a common `shortCircuit` method which took multiple `boolean` parameters.

```
public static <T> boolean shortCircuit(
        T[] array,
        int size,
        Predicate<? super T> predicate,
        boolean expected,
        boolean onShortCircuit,
        boolean atEnd)
{
    for (int i = 0; i < size; i++)
    {
        T each = array[i];
        if (predicate.accept(each) == expected)
        {
            return onShortCircuit;
        }
    }
    return atEnd;
}
```